File: lab2/task1/Main.java
import java.security.SecureRandom;
import java.io.FileWriter;
import java.io.IOException;

/**
 * Utility class for generating random binary sequences
 */
class RandomSequenceGenerator {

    /**
     * Generates 128-bit random binary sequence
     * @return String of 128 bits (0s and 1s)
     */
    public static String generate128bitSequence() {
        SecureRandom random = new SecureRandom();
        byte[] randomBytes = new byte[16];
        random.nextBytes(randomBytes);

        StringBuilder binarySequence = new StringBuilder();
        for (byte b : randomBytes) {
            String binaryString = String.format("%8s", Integer.toBinaryString(b & 0xFF)).replace(' ', '0');
            binarySequence.append(binaryString);
        }

        return binarySequence.toString();
    }

    /**
     * Saves content to a file
     * @param content The content to save
     * @param filename Target filename
     * @throws IOException If file writing fails
     */
    public static void saveToFile(String content, String filename) throws IOException {
        try (FileWriter writer = new FileWriter(filename)) {
            writer.write(content);
        }
    }
}

/**
 * Main class with program entry point
 */
public class Main {
    /**
     * Main method - program entry point
     * @param args Command line arguments (not used)
     */
    public static void main(String[] args) {
        final String FILENAME = "java_sequence.txt";

        try {
            String sequence = RandomSequenceGenerator.generate128bitSequence();
            RandomSequenceGenerator.saveToFile(sequence, FILENAME);
            System.out.println("128-bit sequence saved to " + FILENAME);
            System.out.println("Sequence: " + sequence);
        } catch (IOException e) {
            System.err.println("Error writing to file: " + e.getMessage());
        }
    }
}

File: lab2/task1/main.cpp
#include <iostream>
#include <fstream>
#include <random>
#include <bitset>
#include <string>

using namespace std;

/**
 * Generates a 128-bit pseudorandom binary sequence
 * @return String containing the binary sequence (128 '0' and '1' characters)
 */
string generate_128bit_sequence() {
    random_device rd;
    mt19937_64 generator(rd());

    uint64_t part1 = generator();
    uint64_t part2 = generator();

    string binary_str = bitset<64>(part1).to_string() + bitset<64>(part2).to_string();

    return binary_str;
}

/**
 * Saves string content to a file
 * @param content The string content to save
 * @param filename Target filename
 * @return true if successful, false on error
 */
bool save_to_file(const string& content, const string& filename) {
    ofstream outfile(filename);
    if (!outfile.is_open()) {
        return false;
    }
    outfile << content;
    outfile.close();

    return true;
}

int main() {
    const string filename = "cpp_sequence.txt";

    string sequence = generate_128bit_sequence();

    if (save_to_file(sequence, filename)) {
        cout << "Writed successfully" << endl;
    }
    else {
        cerr << "Error: Writing failed" << endl;
        return 1;
    }

    return 0;
}